ewatercycle/models/abstract.py

```diff
@@ -68,6 +81,38 @@ def set_value(self, name: str, value: np.ndarray) -> None:
         """
         self.bmi.set_value(name, value)
 
+    def set_value_at_coords(self, name: str, lat: Iterable[float], lon: Iterable[float], values: np.ndarray) -> None:
+        """Specify a new value for a model variable at at lat/lon coordinates.
+
+        Args:
+            name: Name of variable
+            lat: Latitudinal value
+            lon: Longitudinal value
+            value: The new value for the specified variable.
+
+        """
+        indices = self._coords_to_indices(name, lat, lon)
+        indices = np.array(indices)
+        self.bmi.set_value_at_indices(name, indices, values)
+
+    def _coords_to_indices(self, name: str, lat: Iterable[float], lon: Iterable[float]) -> Iterable[int]:
+        """Converts lat/lon values to index.
+
+        Args:
+            lat: Latitudinal value
+            lon: Longitudinal value
+
+        """
+        raise NotImplementedError("No generic coords_to_indices method available.")
```

Comment:
Not sure if this error message would be clear to a random user. Perhaps change to something like this? Same for the reverse method.

```suggestion
        raise NotImplementedError("Method to convert from coordinates to model indices not implemented for this model.")
```